### ISSUE TITLE
Add autoformat functionality for source files

### ISSUE THREAD

**Issue by mjdargen** [JuiceMind/JuiceMind#281]
Most modern editors do this, and it makes it a lot easier to read student code (they don't always follow instructions about formatting).

**Comment by juicemind-dev** [comment]
Yes definitely. We were wondering, as a teacher, do you like the idea that students can leverage the auto formatting (they don't have to remember correct formatting because the system does it automatically for them)? I think that's the main reason we have been a bit reluctant to add this feature.

**Comment by mjdargen** [comment]
I could see this varying from teacher to teacher. Personally, I like my students to have the option. I encourage them to follow proper formatting, but then use it at the end to double-check. It exists in all modern editors, and they will use it moving forward outside the class.

**Comment by juicemind-dev** [comment]
Yes, good point. I think we will make it the default on save and add a little button at the top right where they can format their code.

In the future we will add a setting from the teacher side to allow/disallow formatting from the student side.
